src/python/pants/engine/internals/parser.py

```diff
@@ -93,19 +95,30 @@ class Registrar:
             def __init__(self, type_alias: str) -> None:
                 self._type_alias = type_alias
 
-            def __call__(self, **kwargs: Any) -> TargetAdaptor:
+            def __call__(self, **kwargs: Any) -> None:
```

Comment:
There was no need to return something, apparently.

Reply:
JK. We need it for the deprecated CAOF mechanism. I put back a semi-hacky implementation that we can delete in Pants 2.12